impeller/geometry/rect_unittests.cc

```diff
@@ -44,16 +44,200 @@ TEST(RectTest, RectMakeSize) {
+                     const std::string& rect_desc,                //
+                     int64_t px, int64_t py,                      //
+                     const std::string& pt_desc,                  //
+                     Point expected) {
+    // Scalar point inside Scalar rect
+    EXPECT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(Point(px, py)),
+              expected)
+        << "Point(" << pt_desc << ") in " << rect_desc << " Rect";
+    // Scalar point inside Integer rect
+    EXPECT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(Point(px, py)),
+              expected)
+        << "Point(" << pt_desc << ") in " << rect_desc << " IRect";
+    // Integer point inside Scalar rect
+    EXPECT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(IPoint(px, py)),
+              expected)
+        << "IPoint(" << pt_desc << ") in " << rect_desc << " Rect";
+    // Integer point inside Integer rect
+    EXPECT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(IPoint(px, py)),
+              expected)
+        << "IPoint(" << pt_desc << ") in " << rect_desc << " IRect";
+
+    auto nan = std::numeric_limits<Scalar>::quiet_NaN();
+    auto nan_x = Point(nan, py);
+    auto nan_y = Point(px, nan);
+    auto nan_p = Point(nan, nan);
+    // Nan Scalar point inside Scalar and integer rects
+    EXPECT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(nan_x), Point())
+        << "Point(NaN x) in " << rect_desc << " Rect";
+    EXPECT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(nan_y), Point())
+        << "Point(NaN y) in " << rect_desc << " Rect";
+    EXPECT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(nan_p), Point())
+        << "Point(NaN x&y) in " << rect_desc << " Rect";
+    EXPECT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(nan_x), Point())
+        << "Point(NaN x) in " << rect_desc << " Rect";
+    EXPECT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(nan_y), Point())
+        << "Point(NaN y) in " << rect_desc << " Rect";
+    EXPECT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(nan_p), Point())
+        << "Point(NaN x&y) in " << rect_desc << " Rect";
+  };
+
+  // Tests a rectangle using test_one both normally and all variants of
+  // being empty by reversing the lr and tb points.
+  auto test = [&test_one](int64_t l, int64_t t, int64_t r, int64_t b,  //
+                          int64_t px, int64_t py,                      //
+                          const std::string& pt_desc, Point expected) {
+    test_one(l, t, r, b, "non-empty", px, py, pt_desc, expected);
+    test_one(l, t, l, b, "LR empty", px, py, pt_desc, Point());
+    test_one(r, t, l, b, "LR reversed", px, py, pt_desc, Point());
+    test_one(l, t, r, t, "TB empty", px, py, pt_desc, Point());
+    test_one(l, b, r, t, "TB reversed", px, py, pt_desc, Point());
+    test_one(l, t, l, t, "all empty", px, py, pt_desc, Point());
+    test_one(r, b, l, t, "all reversed", px, py, pt_desc, Point());
+  };
+
+  test(100, 100, 200, 200, 100, 100, "UL", Point(0, 0));
```

Comment:
As an example, I have no idea what this means. What does UL mean? 

Given the number of unnamed parameters, I'd also use the `/*l=*/` style.

Reply:
Upper left. I use these abbreviations in a number of places inside the code UL/LL/LR/UR...

Reply:
> Given the number of unnamed parameters, I'd also use the `/*l=*/` style.

I'm not sure what style you are referring to. Are you suggesting that I name all of the parameters? I guess the issue I have with that is that the comments overwhelm the content. Whenever I see that it actually makes the code harder to read and maintain. There are clear symmetries of how I run through the possible combination of test cases that are clear when the data is in columns like this.

Reply:
Can we just write Upper Left then? Maybe the comment ask is not reasonable, but I didn't know what UL meant until now.

Reply:
I pushed a change that comments the parameters to the test invocations using a header row comment and aligning the various parameters under those headings. That should provide some clarity while still making it clear what changes from one test case to the next.

Reply:
And with the new macro to invoke the test case and print the parameters on failure, there are no "descriptor" arguments any more to be confusing. I went ahead and used full names in the comments that describe the cases, so I'm marking this resolved.